modular_robot_simulation/revolve2/modular_robot_simulation/_build_multi_body_systems/_active_hinge_builder.py

```diff
@@ -150,11 +145,13 @@ def build(
         )
 
         tasks = []
-        child = self._module.children[self._module.ATTACHMENT]
+        attachment_point = self._module.attachment_points[self._module.ATTACHMENT]
+        child = self._module.children.get(self._module.ATTACHMENT)
         if child is not None:
             rotation = Quaternion.from_eulers([child.rotation, 0.0, 0.0])
```

Comment:
for all builders: see slack. move the adding of the child rotation the module itself. here, apply everything except for that rotation. apply its own rotation at the start of building a module

Reply:
yes i refractored this into the unbuilt child.
We now pass the position and parent specific orientation into the make_pose function of the unbuilt child, which will make the pose accordingly.